tools/stmo.md

```diff
@@ -1,5 +1,276 @@
+source is slowly replacing the *Presto* data source. *Presto* contains all of
+the data that's exposed via *Athena* and more, but returns query results much
+more slowly.
+
+Other available data sources include *Crash-DB*, *Tiles*, *Sync Stats*, *Push*,
+*Test Pilot*, *ATMO*, and even a *re:dash metadata* which connects to STMO's
+own redash database. It is beyond the scope of this document to list and
+describe all of the data sources, but if you have questions about particular
+data sets, or would like to know if specific data is or can be made available
+in STMO, please inquire in the #datapipeline or #datatools channels on
+irc.mozilla.org.
+
+Creating an Example Dashboard
+-----------------------------
+
+The rest of this document will take you through the process of creating a
+simple dashboard using STMO.
+
+#### Creating A Query
+
+First we must create a query, which can be initiated by clicking on the 'New
+Query' button near the top left of the page. This will bring you to the query
+editing page:
+
+![New Query Page](/../assets/STMO_screenshots/new_query.png "New Query page")
+
+The first query will be counting the number of client ids that we have coming
+from each country, for the top N countries. There is a "client_count" data set
+that is generated from Firefox telemetry pings, so first we check to see if
+that data is available in the *Athena* data source by typing "client_count"
+into the "Search schema..." search box above the schema browser interface to
+the left of the main query edit box. As of this writing, this data set is not
+available in *Athena*, but if you click on the 'Data Source' dropdown and
+change the selection from 'Athena' to 'Presto', you can see that there is a
+main client_count data set (showing up as 'default.client_count') and a number
+of historical client_count data sets (showing up as
+'default.client_count_v<TIMESTAMP>'). Clicking on "default.client_count" in the
+schema browser exposes the columns that are available in the data set. Two of
+the columns are of interest to us for this query: *country* (for obvious
+reasons) and *hll*.
+
+The *hll* column bears some explanation. "hll" stands for
+[HyperLogLog](https://en.wikipedia.org/wiki/HyperLogLog), a sophisticated
+algorithm that allows for the counting of extremely high numbers of items,
+sacrificing a small amount of accuracy in exchange for using much less memory
+than a simple counting structure. The *client_count* data set uses the *hll*
+column for all of its counting functionality. Converting the *hll* value back
+to a regular numeric value requires the use of the following magic SQL
+incantation:
+
+  ```cardinality(merge(cast(hll as HLL)))```
+
+So a query that extracts all of the unique country values and the count for each
+one, sorted from highest count to lowest count looks like this:
+
+  ```SELECT country, cardinality(merge(cast(hll AS HLL))) AS client_count FROM client_count GROUP BY 1 ORDER BY 2 DESC```
```

Comment:
If you format this as follows you'll get syntax highlighting and you can use some whitespace:

```
```sql
SELECT 
  country,
  cardinality(merge(cast(hll AS HLL))) AS client_count
FROM client_count 
GROUP BY 1 
ORDER BY 2 DESC
```.